AdfDataflowFilePrettifier/Prettifier.cs

```diff
@@ -78,36 +75,96 @@ private static (string, string, string) SplitFileIntoInterestingSections(string
                 );
         }
 
-
-        private const char InnerMagicIndicatorCharacter = '\x05'; // We want to be able to distinguish our 'created' \n characters, etc., from the 'real' ones, so that we can reverse the process later.
-        private const char OuterMagicIndicatorCharacter = '\x06'; // And we also want to do this slightly different at the edges of the script.
-        private const char CRMagicIndicatorCharacter = '\x07';    // Due to limitations of kDiff we want to flag CR (\r) characters separately too.
+        private const string BlockStartString = "\n/// PRETTIFIED SCRIPT START MARKER /// DO NOT ALTER THIS LINE ///\n";
+        private const string BlockEndString = "\n/// PRETTIFIED SCRIPT END MARKER /// DO NOT ALTER THIS LINE ///\n";
 
         private static string PrettifyDataFlowScriptDefinition(string scriptLineToPrettify)
         {
-            var prettierLine = scriptLineToPrettify;
-
-            var markedNewLine = OuterMagicIndicatorCharacter + "\n";
-            prettierLine = prettierLine.Replace("\\r\\n", CRMagicIndicatorCharacter + "\n");
-            prettierLine = prettierLine.Replace("\\r", CRMagicIndicatorCharacter + "\r");
-            prettierLine = prettierLine.Replace("\\n", InnerMagicIndicatorCharacter + "\n");
-            prettierLine = prettierLine.Replace("\\t", InnerMagicIndicatorCharacter + "\t");
-                
-            return markedNewLine + prettierLine + markedNewLine;
+            return BlockStartString + PrettifyString(scriptLineToPrettify) + BlockEndString;
         }
 
         internal static string UglifyTextRepresentingPrettifiedDataFlowFile(string fileContents)
         {
-            var uglierFile = fileContents;
+            var readFromIndex = 0;
+            var uglyFileBuilder = new StringBuilder();
+
+            while (true)
+            {
+                var nextBlockIndicies = GetNextBlockIndices(fileContents, readFromIndex);
+                if (nextBlockIndicies == null)
+                {
+                    // No more blocks in the file
+                    uglyFileBuilder.Append(fileContents.Substring(readFromIndex));
+                    break;
+                }
+                var newContentUpToBlockStart = ReadStringBetween(fileContents, readFromIndex, nextBlockIndicies.StartOfBlockStartMarkerIndex);
+                var blockContent = ReadStringBetween(fileContents, nextBlockIndicies.BlockContentStartIndex, nextBlockIndicies.BlockContentEndIndex);
+
+                uglyFileBuilder.Append(newContentUpToBlockStart);
+                uglyFileBuilder.Append(UglifyString(blockContent));
+
+                readFromIndex = nextBlockIndicies.EndOfBlockEndMarkerIndex;
+            }
+
+            return uglyFileBuilder.ToString();
+        }
+

```

Comment:
Can you wrap `GetNextBlockIndices` into `GetAllBlockIndices`, which returns a (potentially empty) collection of non-null Blocks.
Then foreach over those valid blocks, handling `if(!.Any())` upfront by simply returning the input string as-is.

Reply:
I like this idea a lot 👍